src/nimtenbrew/ctr/icon.nim

```diff
@@ -0,0 +1,83 @@
+import ../strings
+
+import strutils
+import strformat
+
+import nimPNG
+import stew/endians2
+
+proc load*(png_path: string, size: int): string =
+    ## Loads a PNG icon at `png_path` with expected size `size`
+    ## Returns the data buffer loaded by nimPNG
+
+    try:
+        let png = nimPNG.loadPNG32(png_path)
+
+        if png.width != size and png.height != size:
+            strings.error(Error.InvalidCtrIconSize)
+
+        return png.data
+    except Exception:
+        return ""
+
+## https://github.com/devkitPro/3dstools/blob/master/src/smdhtool.cpp#L120-L126
+const TILE_ORDER: seq[int] = @[0, 1, 8, 9, 2, 3, 10, 11, 16, 17, 24, 25, 18, 19,
+        26, 27, 4, 5, 12, 13, 6, 7, 14, 15, 20, 21, 28, 29, 22, 23, 30, 31, 32,
+        33, 40, 41, 34, 35, 42, 43, 48, 49, 56, 57, 50, 51, 58, 59, 36, 37, 44,
+        45, 38, 39, 46, 47, 52, 53, 60, 61, 54, 55, 62, 63]
+
+proc convertToRGB565(a, r, g, b: uint8): uint16 =
+    ## Converts ARGB colors to RGB565
+    ## https://github.com/devkitPro/3dstools/blob/master/src/smdhtool.cpp#L274-L285
+
+    var newRed = ((1.float * r.float * a.float) / 255.float).uint16
```

Comment:
You can just do 1.0 and 255.0 to have those automatically become floats - then you don't need the .float cast for those constants